lua/starfall/libs_sv/constraint.lua

```diff
@@ -32,9 +34,18 @@ SF.RegisterType("Constraint", true, false)
 return function(instance)
 local checkpermission = instance.player ~= SF.Superuser and SF.Permissions.check or function() end
 
+local constraint_library = instance.Libraries.constraint
+
+local ent_meta, ewrap, eunwrap = instance.Types.Entity, instance.Types.Entity.Wrap, instance.Types.Entity.Unwrap
+local constr_methods, constr_meta, cwrap, cunwrap = instance.Types.Constraint.Methods, instance.Types.Constraint, instance.Types.Constraint.Wrap, instance.Types.Constraint.Unwrap
+local vwrap, vunwrap = instance.Types.Vector.Wrap, instance.Types.Vector.Unwrap
+local awrap, aunwrap = instance.Types.Angle.Wrap, instance.Types.Angle.Unwrap
+
+
 local getent
 instance:AddHook("initialize", function()
 	getent = instance.Types.Entity.GetEntity
+	constr_meta.__tostring = ent_meta.__tostring
```

Comment:
`starfall/libs_sh/builtins.lua:482: Type mismatch (Expected Entity, got Constraint) in function tostring`

That's because `ent_meta.__tostring` calls `eunwrap` which doesn't handle constraints.